### PR TITLE
Refs #13431 - Apache changes for pulp 2.8

### PR DESCRIPTION
**Pull request by parthaa**

### CODE REVIEW

**Comment by ehelms** [comment]
Can you propogate this knowledge and change over to https://github.com/Katello/puppet-pulp/blob/master/manifests/apache.pp so that it is up to date as well?

File: templates/pulp.conf.erb [hunk 1]
```diff
@@ -17,15 +17,19 @@ SSLInsecureRenegotiation on
...
 #WSGIRestrictStdout Off
 
 WSGISocketPrefix run/wsgi
-WSGIScriptAlias /pulp/api /srv/pulp/webservices.wsgi
-WSGIImportScript /srv/pulp/webservices.wsgi process-group=pulp application-group=pulp
+WSGIScriptAlias /pulp/api /usr/share/pulp/wsgi/webservices.wsgi
+WSGIImportScript /usr/share/pulp/wsgi/webservices.wsgi process-group=pulp application-group=pulp
+
+<Directory /usr/share/pulp/wsgi>
+    Require all granted
```
Comment (ehelms) [inline templates/pulp.conf.erb]: I think you need to account for differences in apache versions: 

https://github.com/Katello/puppet-katello/blob/master/templates/etc/httpd/conf.d/05-foreman.d/katello.conf.erb#L11-L15

File: templates/pulp.conf.erb [hunk 2]
```diff
@@ -38,6 +42,14 @@ WSGIImportScript /srv/pulp/webservices.wsgi process-group=pulp application-group
     SSLVerifyClient optional
 </Files>
 
+Alias /pulp/static /var/lib/pulp/static
+
+<Location /pulp/static>
+    SSLRequireSSL
+    Options +Indexes
+    Require all granted
```
Comment (ehelms) [inline templates/pulp.conf.erb]: I think you need to account for differences in apache versions: 

https://github.com/Katello/puppet-katello/blob/master/templates/etc/httpd/conf.d/05-foreman.d/katello.conf.erb#L11-L15

File: templates/pulp.conf.erb [hunk 3]
```diff
@@ -38,9 +46,25 @@ WSGIImportScript /srv/pulp/webservices.wsgi process-group=pulp application-group
...
+    SSLRequireSSL
+    Options +Indexes
+<%- if scope.function_versioncmp([@apache_version, '2.4']) >= 0 -%>
+     Require all granted
+<% else -%>
+     Allow from all
+<% end -%>
+</Location>
+
 <%- if scope.lookupvar('operatingsystem') == 'Fedora' && scope.lookupvar('operatingsystemrelease') == '18' -%>
```
Comment (ehelms) [inline templates/pulp.conf.erb]: You could just remove lines 61-73

**Comment by jlsherrill** [comment]
Looks good to me i'd open an issue about isolating that configuration in pulp.conf

**Comment by parthaa** [comment]
> Looks good to me i'd open an issue about isolating that configuration in pulp.conf

http://projects.theforeman.org/issues/13486

**Comment by ehelms** [comment]
ACK

**Comment by ehelms** [comment]
Thanks @parthaa

**Pull request merged**